flytekit-java/src/main/java/org/flyte/flytekit/MonoVariableSdkType.java

```diff
@@ -0,0 +1,64 @@
+package org.flyte.flytekit;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+import org.flyte.api.v1.Literal;
+import org.flyte.api.v1.Variable;
+
+/**
+ * Implementation of {@link SdkType} for types that only contain a single variable.
+ * @param <T> Native type for the type of the single variable.
+ */
+class MonoVariableSdkType<T> extends SdkType<SdkBindingData<T>> {
```

Comment:
**Note to reviewers**
This class is package private but if you have a better name please say so

Reply:
I think the [arity](https://en.wikipedia.org/wiki/Arity) concept matches this quite well, so `nullary` (0), `unary` (1), `binary` (2), …, `variadic`(n) are the common prefixes for those.
```suggestion
class UnaryVariableSdkType<T> extends SdkType<SdkBindingData<T>> {
```